backend/otr/src/test/java/fi/oph/otr/service/PublicInterpreterServiceTest.java

```diff
@@ -47,79 +54,88 @@ public void testOnlyPublishingPermittedAreReturned() {
     final LocalDate previousWeek = today.minusDays(7);
     final LocalDate yesterday = today.minusDays(1);
 
-    createLanguagePair(createLegalInterpreter(createInterpreter("not published"), false), "FI", "SE", today, tomorrow);
-    final Kielipari expectedLanguagePair1 = createLanguagePair(
-      createLegalInterpreter(createInterpreter("published1"), true),
-      "FI",
-      "EN",
-      today,
-      tomorrow
-    );
-    final Kielipari expectedLanguagePair2 = createLanguagePair(
-      createLegalInterpreter(createInterpreter("published2"), true),
-      "NO",
-      "FI",
-      yesterday,
-      today
-    );
-    createLanguagePair(
-      createLegalInterpreter(createInterpreterDeleted("deleted1"), true),
-      "DE",
-      "FI",
-      yesterday,
-      today
-    );
-    createLanguagePair(
-      createLegalInterpreterDeleted(createInterpreter("deleted2"), true),
-      "FI",
-      "DE",
-      yesterday,
-      today
-    );
-    createLanguagePair(createLegalInterpreter(createInterpreter("in future"), true), "FI", "DN", tomorrow, nextWeek);
-    createLanguagePair(createLegalInterpreter(createInterpreter("in past"), true), "FI", "IT", previousWeek, yesterday);
-
-    final List<InterpreterDTO> published = publicInterpreterService.list();
-    assertEquals(2, published.size());
-
-    final InterpreterDTO publishedInterpreter1 = published.get(0);
-    assertLanguagePair(expectedLanguagePair1, publishedInterpreter1);
-
-    final InterpreterDTO publishedInterpreter2 = published.get(1);
-    assertLanguagePair(expectedLanguagePair2, publishedInterpreter2);
+    final Tulkki interpreter1 = createInterpreter();
+    final Tulkki interpreter2 = createInterpreter();
+    final Tulkki interpreter3 = createInterpreter();
+    final Tulkki interpreter4 = createInterpreter();
+    final Tulkki interpreter5 = createInterpreterDeleted();
+    final Tulkki interpreter6 = createInterpreter();
+
+    final Oikeustulkki legalInterpreter1 = createLegalInterpreter(interpreter1);
+    final Oikeustulkki legalInterpreter2 = createLegalInterpreter(interpreter2);
+    final Oikeustulkki legalInterpreter3 = createLegalInterpreter(interpreter3);
+    final Oikeustulkki legalInterpreter4 = createLegalInterpreter(interpreter4);
+    final Oikeustulkki legalInterpreter5 = createLegalInterpreter(interpreter5);
+    final Oikeustulkki legalInterpreter6 = createLegalInterpreterDeleted(interpreter6);
+
+    legalInterpreter1.setJulkaisulupaEmail(false);
+    legalInterpreter1.setJulkaisulupaMuuYhteystieto(true);
+    legalInterpreter1.setMuuYhteystieto("oikeustulkki.company@invalid");
+    legalInterpreter2.setJulkaisulupaPuhelinnumero(false);
```

Comment:
Täällä olevat emailin ja puh. numeron julkaisulupien poisto ehkä turhaa testikoodia. Tai nojaavat nyt siihen, että ONR-integraatiota toteutettaessa oppijanumerorekisteristä saadaan näitä tulkkeja koskevat sähköpostiosoitteet ja puhelinnumerot. Nämä voin tosiaan poistaa jos halutaan.